src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -42,13 +56,119 @@ public static int IndexOf(ref byte searchSpace, int searchSpaceLength, ref byte
                     break;  // The unsearched portion is now shorter than the sequence we're looking for. So it can't be there.
 
                 // Found the first element of "value". See if the tail matches.
-                if (SequenceEqual(ref Unsafe.Add(ref searchSpace, offset + 1), ref valueTail, (nuint)valueTailLength))  // The (nuint)-cast is necessary to pick the correct overload
+                if (SequenceEqual(ref Unsafe.Add(ref searchSpace, offset + 1), ref valueTail, valueTailNLength))  // The (nuint)-cast is necessary to pick the correct overload
                     return offset;  // The tail matched. Return a successful find.
 
                 remainingSearchSpaceLength--;
                 offset++;
             }
             return -1;
+
+            // Based on http://0x80.pl/articles/simd-strfind.html#algorithm-1-generic-simd "Algorithm 1: Generic SIMD" by Wojciech Muła
+            // Some details about the implementation can also be found in https://github.com/dotnet/runtime/pull/63285
+        SEARCH_TWO_BYTES:
+            if (Avx2.IsSupported && searchSpaceLength - valueTailLength >= Vector256<byte>.Count)
```

Comment:
@stephentoub I personally not a fan of `IsHardwareAccelerated` property at all but the same applies to `Vector<>.IsHardwareAccelerated` they both return `true` on an e.g. Core i7 Ivy Bridge with AVX1 but most likely use slow paths for many APIs with integers. In IndexOf I rely on AVX2.

Reply:
I maintain that if this can't be successful just using Vector*, something is wrong. A key point of these APIs is to not have to use or understand the low-level intrinsics.

Reply:
cc @tannergooding

Reply:
Like I stated in one of the other comments, I'm fine with us updating `Vector256<T>.IsHardwareAccelerated` to only return `true` on AVX2+ where both floating-point and integer operations can be accelerated using "single instructions".

Reply:
Do we need a follow up issue opened then, to remove use of `Avx2.IsSupported` here? presumably after VectorNNN is fully employed in these files, there should be no using statements for System.Runtime.Intrinsics.Arm nor for System.Runtime.Intrinsics.X86, if I correctly understand the ambitions of the new Vector APIs.

Reply:
> Do we need a follow up issue opened then, to remove use of Avx2.IsSupported here?

Yes, if it's not going to be addressed in this PR, we need to address it subsequently soon.

Reply:
@tannergooding, is this something you're able to follow up on?

Reply:
I opened https://github.com/dotnet/runtime/issues/64309